packages/dds/shared-object-base/src/serializer.ts

```diff
@@ -116,12 +123,11 @@ export class FluidSerializer implements IFluidSerializer {
 
 	// If the given 'value' is an IFluidHandle, returns the encoded IFluidHandle.
 	// Otherwise returns the original 'value'.  Used by 'encode()' and 'stringify()'.
-	private readonly encodeValue = (value: any, bind: IFluidHandle) => {
-		// Detect if 'value' is an IFluidHandle.
-		const handle = value?.IFluidHandle;
-
+	private readonly encodeValue = (value: unknown, bind: IFluidHandleInternal) => {
 		// If 'value' is an IFluidHandle return its encoded form.
-		return handle !== undefined ? this.serializeHandle(handle, bind) : value;
+		return typeof value === "object" && value !== null && fluidHandleSymbol in value
```

Comment:
This is the main non-compile time change: this detection of fluid handles should be much more false positive resistant since valid JSON compatible data should never contain fluidHandleSymbol (since symbols arn't JSON compatible, and that specific symbol should never be used by anything except FluidHandles) while the old code relied on the JSON data not using the string "IFluidHandle" which is a perfectly valid key in JSON data that could be used.